src/uwtools/cli.py

```diff
@@ -42,20 +42,27 @@ def main() -> None:
     # defined checks for the appropriate [sub]mode. Reconfigure logging after quiet/verbose choices
     # are known, then dispatch to the [sub]mode handler.
 
-    setup_logging(quiet=True)
-    args, checks = _parse_args(sys.argv[1:])
-    for check in checks[args[STR.mode]][args[STR.action]]:
-        check(args)
-    setup_logging(quiet=args[STR.quiet], verbose=args[STR.verbose])
-    log.debug("Command: %s %s", Path(sys.argv[0]).name, " ".join(sys.argv[1:]))
-    modes = {
-        STR.config: _dispatch_config,
-        STR.fv3: _dispatch_fv3,
-        STR.rocoto: _dispatch_rocoto,
-        STR.sfcclimogen: _dispatch_sfc_climo_gen,
-        STR.template: _dispatch_template,
-    }
-    sys.exit(0 if modes[args[STR.mode]](args) else 1)
+    try:
+        setup_logging(quiet=True)
+        args, checks = _parse_args(sys.argv[1:])
+        for check in checks[args[STR.mode]][args[STR.action]]:
+            check(args)
+        setup_logging(quiet=args[STR.quiet], verbose=args[STR.verbose])
+    except UWError as e:
+        _abort(str(e))
+    try:
+        log.debug("Command: %s %s", Path(sys.argv[0]).name, " ".join(sys.argv[1:]))
+        modes = {
+            STR.config: _dispatch_config,
+            STR.fv3: _dispatch_fv3,
+            STR.rocoto: _dispatch_rocoto,
+            STR.sfcclimogen: _dispatch_sfc_climo_gen,
+            STR.template: _dispatch_template,
+        }
+        sys.exit(0 if modes[args[STR.mode]](args) else 1)
+    except UWError as e:
+        log.error(str(e))
+        sys.exit(1)
```

Comment:
We removed the `--debug` switch in a recent PR with the understanding that we'd handle exceptions that we recognize and can do something about, letting only exceptions from obvious causes (e.g. `ZeroDivisionError`) or obscure causes (that we might later decide to write handlers for) percolate up.

I realized while implementing this PR's `--env` feature that there are already some cases where we're raising `UWError` exceptions with helpful messages for users, which we can catch here and report without the accompanying stacktrace. For example, on `main` at the moment:
```
% echo "{{ x }}" | uw template render --values-file values.txt
Traceback (most recent call last):
  File "/home/maddenp/git/uwtools/src/uwtools/utils/file.py", line 103, in get_file_format
    return FORMAT.formats()[suffix]
           ~~~~~~~~~~~~~~~~^^^^^^^^
KeyError: 'txt'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/maddenp/sw/conda/envs/DEV-uwtools/bin/uw", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/cli.py", line 48, in main
    check(args)
  File "/home/maddenp/git/uwtools/src/uwtools/cli.py", line 784, in _check_template_render_vals_args
    args[STR.valsfmt] = get_file_format(args[STR.valsfile])
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/utils/file.py", line 107, in get_file_format
    raise UWError(msg) from e
uwtools.exceptions.UWError: Cannot deduce format of 'values.txt' from unknown extension 'txt'
```
Kind of a mess. But since this is _our_ error, and we know we're providing a useful text value in the exception, we'd probably prefer (as implemented in this PR):
```
% echo "{{ x }}" | uw template render --values-file values.txt
Cannot deduce format of 'values.txt' from unknown extension 'txt'
```
So, it seems to me that a good convention might be to catch and handle `UWError` exceptions (which includes subclasses, though we might sometimes catch and handle those further down the call stack) here, letting non-UW exceptions through.